fl4health/clients/basic_client.py

```diff
@@ -47,13 +50,20 @@ def __init__(
                 each batch. Defaults to LossMeterType.AVERAGE.
             checkpointer (Optional[TorchCheckpointer], optional): Checkpointer to be used for client-side
                 checkpointing. Defaults to None.
+            metrics_reporter (Optional[MetricsReporter], optional): A metrics reporter instance to record the metrics
+                during the execution. Defaults to an instance of MetricsReporter with default init parameters.
         """
 
         self.data_path = data_path
         self.device = device
         self.metrics = metrics
         self.checkpointer = checkpointer
 
+        if metrics_reporter is not None:
```

Comment:
Maybe this will be answered as I go through, but does this mean we're always going to be dumping metrics, regardless of whether a reporter is provided?

Reply:
I think I see. This will track metrics, but doesn't necessarily write until `dump` is called. I assume that we'll modify our examples to dump the reporter when you refactor the smoke tests?

Reply:
Yes, correct.